src/components/Reports/GoalCalculator/SharedComponents/GoalCalculatorGrid/GoalCalculatorGrid.tsx

```diff
@@ -0,0 +1,274 @@
+      </Formik>
+    </>
+  );
+};
+
+interface GoalCalculatorGridFormProps {
+  formikProps: FormikProps<GoalCalculatorGridFormValues>;
+  categoryName: string;
+}
+
+const GoalCalculatorGridForm: React.FC<GoalCalculatorGridFormProps> = ({
+  formikProps,
+  categoryName,
+}) => {
+  const { t } = useTranslation();
+  const { values, setFieldValue } = formikProps;
+  const locale = useLocale();
+
+  const totalAmount = values.gridData.reduce(
+    (sum, item) => sum + item.amount,
+    0,
+  );
+
+  const dataWithTotal = [
+    ...values.gridData,
+    { id: 'total', name: 'Total', amount: totalAmount },
+  ];
+
+  const [directInput, setDirectInput] = useState(false);
+
+  const addExpense = () => {
+    const newId = Math.max(...values.gridData.map((item) => item.id), 0) + 1;
+    const newIncomeItem = {
+      id: newId,
+      name: t('New Income'),
+      amount: 0,
+    };
+    const updatedData = [...values.gridData, newIncomeItem];
+    setFieldValue('gridData', updatedData);
+  };
+
+  const handleDirectInputToggle = () => {
+    setDirectInput(!directInput);
+  };
+
+  const handleDelete = (id: number) => {
+    const updatedData = values.gridData.filter((item) => item.id !== id);
+    setFieldValue('gridData', updatedData);
+  };
+
+  const processRowUpdate = (newRow: GridValidRowModel) => {
+    if (newRow.id === 'total') {
+      return newRow;
+    }
+
+    const updatedData = values.gridData.map((item) =>
+      item.id === newRow.id
+        ? {
+            ...item,
+            name: newRow.name as string,
+            amount: newRow.amount as number,
+          }
+        : item,
+    );
+    setFieldValue('gridData', updatedData);
+    return newRow;
+  };
+
+  const columns: GridColDef[] = [
+    {
+      field: 'name',
+      headerName: categoryName,
+      flex: 1,
+      minWidth: 200,
+      editable: true,
+    },
+    {
+      field: 'amount',
+      headerName: t('Amount'),
+      flex: 1,
+      minWidth: 150,
+      editable: true,
+      type: 'number',
+      align: 'center',
+      headerAlign: 'center',
+      renderCell: (params) => currencyFormat(params.value, 'USD', locale),
+    },
+    {
+      field: 'actions',
+      type: 'actions',
+      headerName: '',
+      width: 60,
+      getActions: (params) => {
+        // Don't show delete action for total row
+        if (params.id === 'total') {
+          return [];
+        }
+
+        return [
+          <GridActionsCellItem
+            key="delete"
+            icon={<DeleteIcon />}
+            label="Delete"
+            onClick={() => handleDelete(params.id as number)}
+            showInMenu={false}
+          />,
+        ];
+      },
+    },
+  ];
+
+  return (
+    <Box sx={{ pt: 2 }}>
+      <StyledCard>
+        <StyledControlsBox>
+          <StyledAddButton
+            variant="text"
+            onClick={addExpense}
+            size="small"
+            startIcon={<AddIcon />}
+          >
+            {t(`Add ${categoryName}`)}
```

Comment:
This label needs to be a static string.

```suggestion
            {t('Add {{category}}', { category: categoryName })}
```